src/brc20-bridge/src/state/config.rs

```diff
@@ -0,0 +1,223 @@
+use std::collections::HashSet;
+use std::time::Duration;
+
+use candid::CandidType;
+use ic_exports::ic_cdk::api::management_canister::bitcoin::BitcoinNetwork;
+use ic_stable_structures::stable_structures::Memory;
+use ic_stable_structures::{CellStructure, MemoryId, MemoryManager, StableCell, Storable};
+use serde::Deserialize;
+
+use super::{DEFAULT_DEPOSIT_FEE, DEFAULT_INDEXER_CONSENSUS_THRESHOLD, DEFAULT_MEMPOOL_TIMEOUT};
+use crate::memory::CONFIG_MEMORY_ID;
+
+pub struct Brc20BridgeConfigStorage<M: Memory> {
+    config: StableCell<Brc20BridgeConfig, M>,
+}
+
+impl<M> Brc20BridgeConfigStorage<M>
+where
+    M: Memory,
+{
+    pub fn new(memory: &dyn MemoryManager<M, MemoryId>) -> Self {
+        Self {
+            config: StableCell::new(memory.get(CONFIG_MEMORY_ID), Brc20BridgeConfig::default())
+                .expect("stable memory config initialization failed"),
+        }
+    }
+
+    pub fn get(&self) -> &Brc20BridgeConfig {
+        self.config.get()
+    }
+
+    pub fn set(&mut self, config: Brc20BridgeConfig) {
+        self.config.set(config).expect("failed to set config");
+    }
+
+    pub fn with_borrow_mut<F>(&mut self, f: F)
+    where
+        F: FnOnce(&mut Brc20BridgeConfig),
+    {
+        let mut config = self.config.get().clone();
+
+        f(&mut config);
+
+        self.set(config);
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Deserialize)]
+pub struct Brc20BridgeConfig {
+    pub network: BitcoinNetwork,
+    pub min_confirmations: u32,
+    pub indexer_urls: HashSet<String>,
+    pub deposit_fee: u64,
+    pub mempool_timeout: Duration,
+    /// Minimum quantity of indexer nodes required to reach agreement on a
+    /// request
+    pub indexer_consensus_threshold: u8,
+}
+
+impl Storable for Brc20BridgeConfig {
+    const BOUND: ic_stable_structures::Bound = ic_stable_structures::Bound::Unbounded;
+
+    /* Encoding
+       1                                            // network
+       4                                            // min_confirmations
+       1                                            // number of indexers
+       number of indexers * [1 + indexer_url.len]   // [len] + [indexer_url]
+       8                                            // deposit_fee
+       8                                            // mempool_timeout
+       1                                            // indexer_consensus_threshold
+    */
+
+    fn to_bytes(&self) -> std::borrow::Cow<[u8]> {
+        let mut buf = Vec::with_capacity(
```

Comment:
The effort is remarkable, but it is definitely more maintanable to use candid::Encode here instead